app/acapy_wallet_facade.py

```diff
@@ -0,0 +1,84 @@
+import logging
+from aries_cloudcontroller import AriesAgentControllerBase
+
+from fastapi import HTTPException
+
+logger = logging.getLogger(__name__)
+
+
+async def create_did(controller: AriesAgentControllerBase):
+    """
+    Creates a DID against the ledger using an AriesController
+
+    Parameters:
+    -----------
+    controller: AriesController
+        The aries_cloudcontroller object
+
+    Returns:
+    --------
+    generate_did_response: dict
+        The response object from generating a DID on the ledger
+    """
+    generate_did_res = await controller.wallet.create_did()
+    if not generate_did_res["result"]:
+        logger.error(f"Failed to create DID:\n{generate_did_res}")
+        raise HTTPException(
+            status_code=404,
+            detail=f"Something went wrong.\nCould not generate DID.\n{generate_did_res}",
+        )
+    return generate_did_res
+
+
+async def assign_public_did(controller: AriesAgentControllerBase, did: str):
```

Comment:
changed this method to take a did as a string (not a did object). 

The only thing used by the call is the did so did object is unnecessary abstraction.